File: 0150-evaluate-reverse-polish-notation/0150-evaluate-reverse-polish-notation.py
class Solution:
    def evalRPN(self, tokens: List[str]) -> int:
        stk = []

        for i in range(len(tokens)):
            if tokens[i] == "+" or tokens[i] == "-" or tokens[i] == "*" or tokens[i] == "/":
                operand2 = stk.pop()
                operand1 = stk.pop()
                if tokens[i] == "+":
                    stk.append(operand1 + operand2)
                elif tokens[i] == "-":
                    stk.append(operand1 - operand2)
                elif tokens[i] == "*":
                    stk.append(operand1 * operand2)
                elif tokens[i] == "/":
                    stk.append(int(operand1 / operand2))
            else:
                stk.append(int(tokens[i]))

        return stk[-1]

File: LeetCode/0150-evaluate-reverse-polish-notation/0150-evaluate-reverse-polish-notation.java
public class Solution {
    public int evalRPN(String[] tokens) {
        Stack<Integer> stk = new Stack<>();
        int n = tokens.length;

        Map<String, BiFunction<Integer, Integer, Integer>> mp = new HashMap<>();
        mp.put("+", (a, b) -> a + b);
        mp.put("-", (a, b) -> a - b);
        mp.put("*", (a, b) -> a * b);
        mp.put("/", (a, b) -> a / b);

        for (int i = 0; i < n; i++) {
            if (mp.containsKey(tokens[i])) {
                int operator2 = stk.pop();
                int operator1 = stk.pop();
                int result = mp.get(tokens[i]).apply(operator1, operator2);
                stk.push(result);
            } else {
                stk.push(Integer.parseInt(tokens[i]));
            }
        }

        return stk.pop();
    }
}

/*
class Solution {
    public int evalRPN(String[] tokens) {
        Stack<Integer> stk = new Stack<>();
        int n = tokens.length;
        int result = 0;

        for (int i = 0; i < n; i++) {
            if (tokens[i].equals("+") || tokens[i].equals("-") || tokens[i].equals("*") || tokens[i].equals("/")) {
                int operator2 = stk.pop();
                int operator1 = stk.pop();
                if (tokens[i].equals("+")) {
                    result = operator1 + operator2;
                } else if (tokens[i].equals("-")) {
                    result = operator1 - operator2;
                } else if (tokens[i].equals("*")) {
                    result = operator1 * operator2;
                } else if (tokens[i].equals("/")) {
                    result = operator1 / operator2;
                }
                stk.push(result);
            } else {
                stk.push(Integer.parseInt(tokens[i]));
            }
        }

        return stk.peek();
    }
}
*/